pyproject.toml

```diff
@@ -105,6 +106,30 @@ venvPath = "."
 reportUnreachable = false  # unavoidable with `sys.version_info` conditionals
 
 
+[tool.mypy]
+python_version = "3.9"
+modules = ["mainpy", "tests"]
+exclude = ["^.venv/.*"]
+strict = true
+
+disallow_any_generics = true
+disallow_untyped_calls = true
+disallow_incomplete_defs = true
+disallow_untyped_defs = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+warn_unreachable = false
+# based
+disallow_any_decorated = true
+disallow_any_explicit = true
+disallow_any_expr = true
+disallow_subclassing_any = true
```

Comment:
all of these are default in basedmypy

Reply:
additionally, they aren't exactly based config options, they exist in normal mypy (just without specific error codes), it might be misleading to denote them as based?

Reply:
@KotlinIsland Ah I thought that the ones listed at https://kotlinisland.github.io/basedmypy/error_code_list3.html were based, but I see now that thats limited to some rules that associate to these error codes 👌🏻

Reply:
Are

```toml
disable_bytearray_promotion = true
disable_memoryview_promotion = true
```

also enabled by default?

Reply:
in the docs it says that `strict` is `false` by default 🤔

Reply:
WHAT, fail

Reply:
thanks, i raised them

Reply:
> Are
> 
> ```toml
> disable_bytearray_promotion = true
> disable_memoryview_promotion = true
> ```
> 
> also enabled by default?

So, that's a no?

Reply:
I guess it's a no: KotlinIsland/basedmypy#765